src/Unity.Mathematics/math.cs

```diff
@@ -6471,6 +6471,12 @@ public static void orthonormal_basis(double3 normal, out double3 basis1, out dou
             basis2.z = -normal.y;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        static float4 chgsign(float4 x, float4 sign)
```

Comment:
@unpacklo I think it is worthwhile to make this function public as we do use it at different places.
I'm thinking of bringing in the other variants to, such as:
static float3 chgsign(float3 x, float3 sign)
static float2 chgsign(float2 x, float2 sign)
static float chgsign(float x, float sign)

With new tests. Should these all live in the math namespace too ?

Reply:
@unpacklo have a look at https://github.com/Unity-Technologies/Unity.Mathematics/pull/201/commits/36527fb7389c6a943dcbb537bb911cbcb881445e

Reply:
@oldio Yes, these should be public and we should have equivalents for all the different vector lengths. All of these will go in the math namespace. I can go ahead and start implementing those along with the tests.

Reply:
Haha, I think we posted a the same time :)